packages/core/src/test/login/webview/vue/backend_toolkit.test.ts

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SinonSandbox, createSandbox } from 'sinon'
+import { assertTelemetry, tryRegister } from '../../../testUtil'
+import assert from 'assert'
+import { createTestAuth } from '../../../credentials/testUtil'
+import { Auth } from '../../../../auth'
+import { isBuilderIdConnection, isIdcSsoConnection, scopesSsoAccountAccess } from '../../../../auth/connection'
+import { getOpenExternalStub } from '../../../globalSetup.test'
+import { openAmazonQWalkthrough } from '../../../../amazonq/onboardingPage/walkthrough'
+import { ToolkitLoginWebview } from '../../../../login/webview/vue/toolkit/backend_toolkit'
+import {
+    CodeCatalystAuthenticationProvider,
+    CodeCatalystAuthStorage,
+    defaultScopes,
+} from '../../../../codecatalyst/auth'
+import { FakeSecretStorage, FakeMemento } from '../../../fakeExtensionContext'
+import * as authUtils from '../../../../auth/utils'
+
+// TODO: remove auth page and tests
+describe('Toolkit Login', function () {
+    const region = 'fakeRegion'
+    const startUrl = 'fakeUrl'
+    const profileName = 'profile'
+    const accessKey = 'fakeKey'
+    const secretKey = 'fakeSecret'
+
+    let sandbox: SinonSandbox
+    let auth: ReturnType<typeof createTestAuth>
+    let codecatalystAuth: CodeCatalystAuthenticationProvider
+    let backend: ToolkitLoginWebview
+
+    before(function () {
+        tryRegister(openAmazonQWalkthrough)
+    })
+
+    beforeEach(function () {
+        sandbox = createSandbox()
+        auth = createTestAuth()
+        codecatalystAuth = new CodeCatalystAuthenticationProvider(
+            new CodeCatalystAuthStorage(new FakeSecretStorage()),
+            new FakeMemento(),
+            auth
+        )
+        sandbox.stub(Auth, 'instance').value(auth)
+        getOpenExternalStub().resolves(true)
+
+        backend = new ToolkitLoginWebview(codecatalystAuth)
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    it('successfully signs into builder ID and emits telemetry', async function () {
+        sandbox.stub(codecatalystAuth, 'isConnectionOnboarded').resolves(true)
+
+        await backend.startBuilderIdSetup()
+
+        assert.ok(isBuilderIdConnection(auth.activeConnection))
+        assert.deepStrictEqual(auth.activeConnection.scopes, defaultScopes)
+        assert.deepStrictEqual(auth.activeConnection.state, 'valid')
+
+        assertTelemetry('auth_addConnection', {
+            result: 'Succeeded',
+            credentialSourceId: 'awsId',
+            authEnabledFeatures: 'codecatalyst',
+            isReAuth: false,
+        })
+    })
+
+    it('successfully signs into account IdC and emits telemetry', async function () {
```

Comment:
"successfully" is usually redundant in test names. It would be unusual to test that something failed (and then we would say "fails" in the name)
```suggestion
    it('signs into account IdC and emits telemetry', async function () {
```